<launch>
    <!-- model predictive controller -->
    <node pkg="barc" type="controller_MPC.jl" name="MPC" output="screen">
    </node>

	<!-- inertial measurement unit -->
    <node pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud_rate" value="57600" />
    </node>

	<!-- arduino nano -->
    <node pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node>

    <!-- state observer -->
    <node pkg="barc" type="state_estimation_KinBkMdl.py" name="state_estimation" output="screen">
        <!-- vehicle properties -->
        <param name="L_a" type="double" value="0.125" />
        <param name="L_b" type="double" value="0.125" />
        
        <!-- v_x estimation sample time-->
        <param name="dt_vx" type="double" value="0.2" />

        <!-- ekf properties -->
        <param name="q_std" type="double" value="0.1" />
        <param name="r_std" type="double" value="0.1" />

        <param name="input_gain" type="double" value="0.4" />
    </node>

</launch>
